lib/protobuf/field/base_field.rb

```diff
@@ -136,6 +144,16 @@ def required?
       # FIXME: need to cleanup (rename) this warthog of a method.
       def set(message_instance, bytes)
         return message_instance[name] = decode(bytes) unless repeated?
+
+        if map?
+          hash = message_instance[name]
+          entry = decode(bytes)
+          # decoded value could be nil for an
+          # enum value that is not recognized
+          hash[entry.key] = entry.value unless entry.value.nil?
+          return
```

Comment:
This should be `return entry.value` right? It looks like the semantics for this method is it always returns the set value, so we should be consistent.